dace/codegen/cppunparse.py

```diff
@@ -766,7 +811,18 @@ def _Num(self, t):
                 self.write(
                     "%s(0, %s)" % (dtype, repr_n.replace("inf", INFSTR)[:-1]))
             else:
-                self.write(repr_n.replace("inf", INFSTR))
+                self.write(repr_n.replace("inf", INFSTR), infer_type)
+                # If the number has a type, use it
+                if isinstance(t.n, np.uint):
+                    return dace.dtypes.typeclass(np.uint32) if infer_type else None
```

Comment:
@TizianoDeMatteis all of these `dace.dtypes.typeclass(np.*)` can just be replaced by `dace.dtypes.*`